### DISCUSSION TITLE
Please add language switcher

### DISCUSSION THREAD

**Discussion by grothkopp** [ChristopherKlay/StadiaEnhanced#122] [Ideas & Requests]
Hi,

I just learned that you can force stadia to another language like this:
https://stadia.google.com/home?hl=en
obviously "en" is the language code.
Please add a switcher for languages to the StadiaEnhanced settings. That would make life a lot easier for people like me who like to play games in another language.

**Comment by ChristopherKlay** [comment]
Converted into a discussion under feature requests - please leave "issues" for bugs and errors within the extension.

I would definitely like to implement this and have thought about it in the past, the only issue is the amount of available languages and the selection presented to the specific user.

A list with every single supported language wouldn't make much sense, because navigating it would be drastically slower. A simple by-visit functionality also wouldn't work well, because clicking a single link with a given language included, would then include that language in your selection from now on. I will need to figure out a quick way to give each user the ability to create their own list first.

I will add it to the project board for [feature suggestions](https://github.com/ChristopherKlay/StadiaEnhanced/projects/1). (:

**Reply by grothkopp** [reply]
Hi, sorry about opening an issue, I'm not using github that much so I didn't know where to put this.
Maybe a short list of the users own language (browser/request language) and the top common languages (english, french, spanish, ... ) would be enough for a start?
The only usecase I can imagine apart from switching between my native language and english, are countries with multiple languages like Canada or Switzerland. 
If jquery is available for the extension you could also take a look at this: https://harvesthq.github.io/chosen/ It's a searchable dropdown which can handly really long lists very well.

**Reply by ChristopherKlay** [reply]
> Hi, sorry about opening an issue, I'm not using github that much so I didn't know where to put this.

Everything is fine (:

> Maybe a short list of the users own language (browser/request language) and the top common languages (english, french, spanish, ... ) would be enough for a start?

Would definitely be better than nothing.

I will most likely work out a by-visit system that always has "Automatic" and "English" available at the top and then a list of language codes that expands based on you visiting Stadia with them. I.e. if you visit it in french, it would add the french code to the list and so on. (:

**Reply by grothkopp** [reply]
Here is a list of valid values for the hl parameter (at least for google search, I'm not sure if it applies here as well):
https://developers.google.com/custom-search/docs/xml_results_appendices#interfaceLanguages

**Reply by ChristopherKlay** [reply]
Sadly Stadia currently doesn't support most of those. Otherwise it would be fairly easy to just implement a full list.

Your feature request is already implemented in the dev build right now, like this:
![image](https://user-images.githubusercontent.com/12472611/114552631-17365a00-9c65-11eb-90b8-43c2bee7b30a.png)

I will add some more common languages as well and it will be part of the next update (:
